enum.rb

```diff
@@ -0,0 +1,115 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Style/CaseEquality
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = instance_of?(Array) ? self : to_a
+    count = 0
+    while count < arr.length
+      yield(arr[count])
+      count += 1
+    end
+    arr
+  end
```

Comment:
- [ ] when a block is given when self is a range returns the range itself after calling the given block once for each element in self

```Ruby
range.my_each(&block)

# expected: 5..50
# got: [5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]
```